src/layout/dom.ts

```diff
@@ -235,17 +245,47 @@ function layout(tag: string, id: number, parentId: number): void {
                 }
                 break;
             case "IMG":
+                value.metadata.boxmodel = true;
+            case "VIDEO":
+            case "AUDIO":
+            case "LINK":
+                // Track mapping between URL and corresponding nodes
+                if (Constant.HREF_ATTRIBUTE in attributes && attributes[Constant.HREF_ATTRIBUTE].length > 0) {
+                    urlMap[getFullUrl(attributes[Constant.HREF_ATTRIBUTE])] = id;
```

Comment:
What if same url maps to multiple ids?

Reply:
The last one based on mutation will win. We have 1:1 right now, no support for 1 to many. Let me know if you have thoughts on how to change it for future version?